docs/release/controllers.md

```diff
@@ -0,0 +1,181 @@
+and a support window of one year will be provided for the previous major version.
+
+## Release cadence
+
+Flux controllers are at least released at the same rate as Kubernetes, following their cadence of three
+minor releases per year. After each Kubernetes minor release, all controllers are tested against the latest
+Kubernetes version and are released at approximately two weeks after Kubernetes.
+The newly released controllers offer support for Kubernetes N-2 minor versions.
+
+A Flux controller may have more than three minor releases per year, if maintainers decide to ship a 
+new feature or optimisation ahead of schedule.
+
+## Supported releases
+
+For Flux controllers we support the last three minor releases.
+
+Security fixes, may be back-ported to those three minor versions as patch releases,
+depending on severity and feasibility.
+
+Note that back-porting is provided by the community on a best-effort basis.
+
+## Release artifacts
+
+Each controller release produces the following artifacts:
+
+- Source code (GitHub Releases page)
+- Software Bill of Materials in SPDX format (GitHub Releases page)
+- Kubernetes manifests such as CRDs and Deployments (GitHub Releases page)
+- Signed checksums of source code, SBOM and manifests (GitHub Releases page)
+- Multi-arch container images (GitHub Container Registry and DockerHub)
+
+All the artifacts are cryptographically signed and can be verified with Cosign.
+
+The release artifacts can be accessed based on the controller name and version.
+
+To import or update a controller's API package in a Go project:
+
+```shell
+go get github.com/fluxcd/<controller-name>/api@<version>
+```
+
+To verify and pull a controller's container image:
+
+```shell
+cosign verify ghcr.io/fluxcd/<controller-name>:<version>
+docker pull ghcr.io/fluxcd/<controller-name>:<version>
+```
+
+To download a controller's Kubernetes Custom resource definitions:
+
+```shell
+curl -sL https://github.com/fluxcd/<controller-name>/releases/download/<version>/<controller-name>.crds.yaml
+```
+
+## Controller release procedure
+
+As a project maintainer, to release a controller and its API:
+
+1. Checkout the `main` branch and pull changes from remote.
+2. Create a new branch from `main` i.e. `release-<next semver>`. This
+   will function as your release preparation branch.
+3. Update the `github.com/fluxcd/<NAME>-controller/api` version in `go.mod`
+4. Add an entry to the `CHANGELOG.md` for the new release and change the
+   `newTag` value in ` config/manager/kustomization.yaml` to that of the
+   semver release you are going to make. Commit and push your changes.
+5. Create a PR for your release branch and get it merged into `main`.
+6. Create a `api/<next semver>` tag for the merge commit in `main` and push it to remote.
+7. Create a `<next semver>` tag for the merge commit in `main` and push it to remote.
+8. Confirm CI builds and releases the newly tagged version.
+
```

Comment:
It might be useful to add some information about handling backporting changes e.g. branch naming conventions, cherry-picking commits. Not sure where the right place for that would be.